source/common/http/http1/conn_pool.cc

```diff
@@ -220,6 +229,11 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
 
 void ConnPoolImpl::onUpstreamReady() {
   upstream_ready_enabled_ = false;
+  while (!delayed_clients_.empty()) {
+    ActiveClient& client = *delayed_clients_.front();
+    client.delayed_ = false;
+    client.moveBetweenLists(delayed_clients_, ready_clients_);
```

Comment:
Not sure if it could happen: 
time X: in batch 0: client A is put into delayed list, setup the timer 
time X+1: in batch 1: client B is put into delayed list
time X+2: in batch 1: both A and B are moved out of the delayed list